File: 34 Find First and Last Position of Element in Sorted Array/34find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        if target not in nums:
            return [-1, -1]
        i = nums.index(target)
        j = i

        while j < n and nums[j] == target:
            j += 1

        return [i, j - 1]


        # if n == 0:
        #     return [-1, -1]
        # idx = (n - 1) // 2

        # while idx > 0 and idx != n - 1 and nums[idx] != target:
            
        #     if nums[idx] > target:
        #         n = idx
        #         idx = idx // 2

        #     elif nums[idx] < target:
        #         idx = (idx + n) // 2
        
        # i, j = idx, idx

        # print (idx)
        # if nums[i] != target and nums[j] != target:
        #     return [-1, -1]

        # while i >= 0 and nums[i] == target:
        #     i -= 1
        # while j < n and nums[j] == target:
        #     j += 1

        # return [i + 1, j - 1]
        

File: 34 Find First and Last Position of Element in Sorted Array/34find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        auto i = find(nums.begin(), nums.end(), target);
        vector<int> res = {-1, -1};

        if (i != nums.end()){
            int j = i - nums.begin();
            res[0] = i - nums.begin();
            while(j < nums.size() && nums[j] == target){
                j++;
            }
            res[1] = j - 1;

        }   
        return res;
    }
};